### DISCUSSION TITLE
Balena won't resume WiFi

### DISCUSSION THREAD

**Discussion by kk-sonata** [ketilmo/balena-ads-b#81] [Q&A] [answered]
If my feeder loses its WiFi connection, it seems to go into WiFi connect mode (Balena hotspot) and won't reconnect automatically. I've never used Balena before, is there a setting I need to change? It's a RPI4. Thanks!!

**Comment by ketilmo** [comment]
Hi @kk-sonata,

Sorry for not getting back to you sooner. When WiFi Connect starts up, it's often a symptom that your device is either missing a connection to the WiFi router, Internet or both. WiFi Connect will automatically disable if the connection is restored. 

Things to try: Move the device closer to the wifi router, or (temporarily) connect the device to your router using an ethernet cable.

Let us know if you make progress on this.

**Reply by kk-sonata** [reply, marked as answer]
Thanks. It seems that the default ACTIVITY_TIMEOUT setting for Wifi Connect is 0 (infinity). So once it starts an AP it will never stop waiting for someone to enter wifi credentials. I've changed it to a more reasonable setting (120).

Understand it's better to use Ethernet, but it should be able to deal with occasionally losing wifi without the need for manual intervention. Hopefully this setting change solves my issue.

**Comment by ketilmo** [comment]
Thank you so much for debugging this, @kk-sonata! I've looked at the code, and it looks like wifi-connect will stay active forever unless specified. To mitigate this, I have added a variable `ACTIVITY_TIMEOUT` with a value of `120` to `docker-compose.yml`: https://github.com/ketilmo/balena-ads-b/blob/d6ed449dcac73a5ee0870373807748a067e06b65/docker-compose.yml#L23

Did the change solve the issue on your end?

**Reply by kk-sonata** [reply]
Yes my setup has been completely stable since I added that variable in my own docker-compose. Thanks!

**Reply by ketilmo** [reply]
Thanks for confirming, @kk-sonata! I have deployed the changes to the main branch. New installations should now have get the correct values by default.
